vite.config.js

```diff
@@ -2,48 +2,13 @@ import { defineConfig } from 'vite';
 import preact from '@preact/preset-vite';
 import { viteStaticCopy } from 'vite-plugin-static-copy';
 import replace from '@rollup/plugin-replace';
-import yaml from 'yaml';
 import path from 'path';
-import { promises as fs } from 'fs';
-import { Readable } from 'stream';
-import { Feed } from 'feed';
-import config from './src/config.json';
 
-/**
- * @param {string} content
- * @param {string} path
- * @returns {string}
- */
-function compileMarkdown(content, path) {
-	// Find YAML FrontMatter preceeding a markdown document
-	const FRONT_MATTER_REG = /^\s*---\n\s*([\s\S]*?)\s*\n---\n/i;
-
-	// Find a leading title in a markdown document
-	const TITLE_REG = /^\s*#\s+(.+)\n+/;
-
-	const matches = content.match(FRONT_MATTER_REG);
-	if (matches) {
-		let meta;
-		try {
-			meta =
-				yaml.parse('---\n' + matches[1].replace(/^/gm, '  ') + '\n') ||
-				{};
-		} catch (e) {
-			throw new Error(`Error parsing YAML FrontMatter in ${path}`);
-		}
-		content = content.replace(FRONT_MATTER_REG, '');
-		if (!meta.title) {
-			let [, title] = content.match(TITLE_REG) || [];
-			if (title) {
-				meta.title = title;
-			}
-		}
-
-		content = '---\n' + JSON.stringify(meta) + '\n---\n' + content;
-	}
-
-	return content;
-}
+import { precompileMarkdown } from './plugins/precompile-markdown/index.js';
+import { netlifyPlugin } from './plugins/netlify.js';
+import { spaFallbackMiddlewarePlugin } from './plugins/spa-fallback-middleware.js';
+import { htmlRoutingMiddlewarePlugin } from './plugins/html-routing-middleware.js';
+import { rssFeedPlugin } from './plugins/rss-feed.js';
```

Comment:
Sorry for the busy diff, the config file was getting a bit big as I kept shoving more plugins/middlewares in. Needed to be split out